Ginger/Ginger/Variables/AddVariablePage.xaml.cs

```diff
@@ -93,69 +144,179 @@ private void XSharedRepoTabListView_MouseDoubleClick(object sender, System.Windo
+
+                Button addParameterBtn = new();
+                addParameterBtn.Content = "Add Parameter";
+                addParameterBtn.Click += new RoutedEventHandler(AddVariableButton_Click);
+
+                Button addParameterBtnToAllEn = new();
+                addParameterBtnToAllEn.Content = "Add Parameter to all Environments";
+                addParameterBtnToAllEn.Click += new RoutedEventHandler(AddVariablesToAllTheEnvironmentsButton_Click);
+                buttons.Add(addParameterBtnToAllEn);
+                buttons.Add(addParameterBtn);
+            }
+
+            else
+            {
+                this.Title = "Add " + GingerDicser.GetTermResValue(eTermResKey.Variable);
+                Button addVarBtn = new Button();
+                addVarBtn.Content = "Add " + GingerDicser.GetTermResValue(eTermResKey.Variable);
+                addVarBtn.Click += new RoutedEventHandler(AddVariableButton_Click);
+                buttons.Add(addVarBtn);
+            }
 
             this.Width = 400;
             this.Height = 400;
 
-            GingerCore.General.LoadGenericWindow(ref _pageGenericWin, App.MainWindow, windowStyle, this.Title, this, new ObservableList<Button> { addVarBtn });
+            GingerCore.General.LoadGenericWindow(ref _pageGenericWin, App.MainWindow, windowStyle, this.Title, this, buttons);
         }
 
         private void AddVariableButton_Click(object sender, RoutedEventArgs e)
         {
+            bool isVariableAdded = false;
             if (xLibraryTab.IsSelected)
             {
-                AddLibraryVariables();
+                isVariableAdded = AddLibraryVariables();
             }
             else
             {
-                AddSharedRepoVariables();
+                isVariableAdded = AddSharedRepoVariables();
             }
 
+            if (isVariableAdded)
+            {
+                _pageGenericWin.Close();
+            }
+        }
+
+
+        /// <summary>
+        /// ONLY for Environment Page
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void AddVariablesToAllTheEnvironmentsButton_Click(object sender, RoutedEventArgs e)
+        {
+            VariableBase varToAdd = (VariableBase)xLibraryTabListView.xListView.SelectedItem;
+            string Name = variableName.Text;
+            string Description = variableDescription.Text;
+            string? Value = varToAdd is VariableDateTime ? dtpInitialDate.Value.ToString() : variableValue.Text;
+
+            varToAdd.Name = Name;
+            varToAdd.Description = Description;
+            varToAdd.SetInitialValue(Value);
+
+            ObservableList<ProjEnvironment> ProjEnvironments = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>();
+
+            ProjEnvironments.ForEach((projEnv) =>
+            {
+                projEnv.StartDirtyTracking();
+
+                projEnv.Applications.ForEach((envApp) =>
```

Comment:
apply where to select only relevant application

Reply:
Implemented this